magma/t.py

```diff
@@ -1,28 +1,40 @@
+import warnings
+import enum
+from abc import abstractmethod
 from .ref import Ref, AnonRef, DefnRef, InstRef
 from .port import INOUT, INPUT, OUTPUT
 from .compatibility import IntegerTypes, StringTypes
 
-__all__  = ['Type', 'Kind']
-__all__ += ['In', 'Out', 'InOut', 'Flip']
+
+# From http://code.activestate.com/recipes/391367-deprecated/
+def deprecated(func):
+    """This is a decorator which can be used to mark functions
+    as deprecated. It will result in a warning being emmitted
+    when the function is used."""
+    def newFunc(*args, **kwargs):
+        warnings.warn("Call to deprecated function %s." % func.__name__,
```

Comment:
Probably want `warnings.warn(msg, category, stacklevel=2)` which makes the warning appear at the call site.

Reply:
Fixed